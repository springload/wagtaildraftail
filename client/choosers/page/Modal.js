import React from 'react';
import PropTypes from 'prop-types';
import { Portal } from 'draftail';

const propTypes = {
  onModalClose: PropTypes.func.isRequired,
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

/**
 * A modal dialog following Wagtail's styles one to one.
 */
class Modal extends React.Component {
  render() {
    const { onModalClose, children } = this.props;
    return (
      <Portal>
        <div
          className="modal fade in"
          tabIndex={-1}
          role="dialog"
          aria-hidden={false}
          style={{ display: 'block' }}
        >
          <div className="modal-dialog">
            <div className="modal-content">
              <button
                onClick={onModalClose}
                type="button"
                className="button close icon text-replace icon-cross"
                data-dismiss="modal"
                aria-hidden={false}
              >
                &times;
              </button>
              <div className="modal-body">
                {children}
              </div>
            </div>
          </div>
        </div>
        <div className="modal-backdrop fade in" />
      </Portal>
    );
  }
}

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;
