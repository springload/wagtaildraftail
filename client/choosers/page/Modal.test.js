import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

describe('Modal', () => {
  it('renders', () => {
    expect(shallow((
      <Modal
        onModalClose={() => {}}
      />
    ))).toMatchSnapshot();
  });
});
