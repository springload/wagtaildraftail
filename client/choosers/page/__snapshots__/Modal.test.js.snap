// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<div>
  <div
    aria-hidden={true}
    className="modal fade in"
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
    tabIndex={-1}
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <button
          aria-hidden={true}
          className="button close icon text-replace icon-cross"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Ã—
        </button>
        <div
          className="modal-body"
        >
          <div>
            Empty
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-backdrop fade in"
  />
</div>
`;
